src/metabase/sync/interface.clj

```diff
@@ -44,6 +44,8 @@
    [:nfc-path                   {:optional true} [:any]]
    [:custom                     {:optional true} :map]
    [:database-is-auto-increment {:optional true} :boolean]
+   ;; nullable for databases that don't support field partition
+   [:database-partitioned       {:optional true} [:maybe :boolean]]
```

Comment:
Why did you decide to make this nullable rather than just optional?

Reply:
no strong reason, indeed we can make it tighter and remove the maybe.

4c19a94e14

Reply:
This schema not only being used for metadata from jdbc but also for field getting from `metabase_field`, see https://github.com/metabase/metabase/blob/4cf4e789cdaf47696db35f8630cb941a5edd17d1/src/metabase/sync/sync_metadata/fields/common.clj#L17

Reply:
Sorry I realised the schema was the wrong place to put this comment. I've restated the question [here](https://github.com/metabase/metabase/pull/36892/files#diff-e6ec4fc8084d80ae2ebedde98c98f358ec7ac526de5c339ebcb454cbda62b221R4396)